src/pages/resultsView/cnSegments/CNSegments.tsx

```diff
@@ -1,139 +1,114 @@
-                this.activeTabId = id;
-            }}>
-                {
-                    this.props.store.genes.result!.map((gene:Gene)=>{
-                        return <MSKTab key={gene.hugoGeneSymbol} id={`CNSegmentsTab${gene.hugoGeneSymbol}`} linkText={gene.hugoGeneSymbol}>
-                            <CNASegmentIframe studyId={this.props.store.studyIds.result![0]}
-                                              sampleIds={this.props.store.samples.result!.map((sample:Sample)=>sample.sampleId)}
-                                              gene={gene}
-                            />
-                        </MSKTab>
-                    })
-                }
-            </MSKTabs>);
+    public render() {
+
+        if (this.props.store.cnSegments.isComplete)
+        {
+            return (
+                <div className="pillTabs">
+                    <DefaultTooltip
+                        overlay={<span>Download a copy number segment file for the selected samples</span>}
+                        placement="left"
+                    >
+                        <Button
+                            className="btn btn-sm pull-right"
+                            onClick={this.handleDownload}
+                        >
+                            <i className='fa fa-cloud-download'/>
+                        </Button>
+                    </DefaultTooltip>
+                    <Nav
+                        bsStyle="pills"
+                        activeKey={this.activeLocus}
+                        onSelect={(id: any) => {
+                             this.selectedLocus = id;
+                        }}
+                    >
+                        <NavItem eventKey={WHOLE_GENOME}>
+                            Whole Genome
+                        </NavItem>
+                        {
+                            this.props.store.genes.result && this.props.store.genes.result.map((gene: Gene) => (
+                                <NavItem eventKey={gene.hugoGeneSymbol}>
+                                    {gene.hugoGeneSymbol}
+                                </NavItem>
+                            ))
+                        }
+                    </Nav>
+                    <IntegrativeGenomicsViewer
+                        tracks={[
+                            {
+                                ...defaultSegmentTrackProps(),
+                                height: this.segmentTrackHeight,
+                                features: this.features
+                            }
+                        ]}
+                        locus={this.activeLocus}
+                    />
+                </div>
+            );
         } else {
-            return null
+            return <LoadingIndicator isLoading={true} center={true} size="big" />;
         }
-
-
     }
 
+    @autobind
+    private handleDownload() {
+        onMobxPromise(
```

Comment:
should we have an alias for onMobxPromise called onMobxPromiseOnce?  This is probably the default use case for it (times=1) and the idea that it disposes after one call is key.  i had to check to make sure that it was disposing and not just firing every time the observed promise resolved.  @adamabeshouse

Reply:
you mean just for readability? bc the default param value is already 1